source/fundamentals/crud/read-operations/limit.txt

```diff
@@ -4,10 +4,166 @@ Limit the Number of Returned Results
 
 .. default-domain:: mongodb
 
-Use ``limit()`` to cap the number of documents that a read operation returns.
-This instance method designates the maximum number of
-documents that a read operation can return. If there are not enough documents
-to reach the specified limit, it can return a smaller number.
-If you use ``limit()`` with the ``skip()`` (TODO: link) instance method, the skip applies
-first and the limit only applies to the documents left over after
-the skip.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to limit the number of documents
+returned from a read operation.
+
+Sample Data
+~~~~~~~~~~~
+
+Run the following snippet to load the documents into the ``ratings``
+collection of the ``tea`` database:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/limit.go
+   :language: go
+   :dedent:
+   :start-after: begin insertDocs
+   :end-before: end insertDocs
+
+.. tip:: Non-existent Databases and Collections
+
+   The driver automatically creates the necessary database and
+   collection when you perform a write operation against them if they
+   don't already exist.
+
+Each document contains a rating for a type of tea that corresponds to
+the ``type`` and ``rating`` fields.
+
+.. note::
+
+   Each example truncates the ``ObjectID`` value because the driver
+   generates them uniquely.
+
+Limit
+-----
+
+To limit the number of documents returned from a query, pass the
+number of documents you want returned to the ``SetLimit()`` function of
+the read operations' options.
+
+Specify the options as the last parameter to the following read
+operation functions:
+
+- ``Find()``
+- ``CountDocuments()``
+- ``gridfs.Bucket.Find()``
+
+If the number of documents is ``0`` or exceeds the number of matched
+documents, the query returns all the documents. 
+
+.. tip::
+
+    Passing in a negative number to the ``SetLimit()`` function is
+    equivalent to passing in the absolute value of the negative number.
```

Comment:
This is true, but also a [negative limit](https://docs.mongodb.com/manual/reference/method/cursor.limit/#negative-values) closes the cursor after retrieving the documents (unlike a positive limit, which leaves the cursor open on the server). We use a limit of `-1` for our [`FindOne`](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/collection.go#L1302) function.